library/psa_crypto.c

```diff
@@ -3609,6 +3609,42 @@ psa_status_t psa_cipher_decrypt( mbedtls_svc_key_id_t key,
 /* AEAD */
 /****************************************************************/
 
+/* Helper to perform common nonce length checks. */
+static psa_status_t psa_aead_check_nonce_length( psa_algorithm_t alg,
+                                                 size_t nonce_length )
+{
```

Comment:
Shouldn't we do here something like: 
```
alg = psa_aead_get_base_algorithm( alg );
```

Reply:
Good catch. Multipart stores the base alg, oneshot does not.

Reply:
Well-spotted. Since this is fixing a bug, it needs a changelog entry, a non-regression test and a backport. I'll push a branch with some tests in a minute.

Reply:
On closer inspection, `mbedtls_gcm_starts` and `mbedtls_ccm_starts` were successful as a second line of defense. The bug was a very minor one: inconsistently returning `INVALID_ARGUMENT` vs `NOT_SUPPORTED`. So this doesn't need a changelog entry. We should still add tests, preferably in this PR — I've pushed some to https://github.com/gilles-peskine-arm/mbedtls/tree/psa_aead_check_nonce_length-tests and I'll backport the one-shot ones.

Reply:
While at it, what about to make this function more robust by moving to whitelist approach rather than a blacklist one:
```
if( base_alg == PSA_ALG_GCM )
{
    if( nonce_length != 0 )
        return( PSA_SUCCESS );
}
...
return( PSA_ERROR_NOT_SUPPORTED );
```